src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -628,7 +628,7 @@ StorageReplicatedMergeTree::StorageReplicatedMergeTree(
                   * Otherwise `metadata_version` for not first replica will be initialized with 0 by default.
                   */
                 setInMemoryMetadata(metadata_snapshot->withMetadataVersion(metadata_version));
```

Comment:
After looking into this again, I have another idea/question, maybe clearing the cache in `setInMemoryMetadata` will be less error prone? Not sure that it make sense for this fix, but interesting what you think